tests/server.py

```diff
@@ -283,12 +283,24 @@ def _install_handlers(self) -> None:
         self._on_notification("exit", self._on_exit)
 
         self._on_request("$test/getReceived", self._get_received)
+        self._on_request("$test/sendNotification", self._on_send_notification)
         self._on_notification("$test/setResponse", self._on_set_response)
 
     async def _on_set_response(self, params: PayloadLike) -> None:
         if isinstance(params, dict):
             self._responses[params["method"]] = params["response"]
 
+    async def _on_send_notification(self, params: PayloadLike) -> None:
+        if not isinstance(params, dict):
+            raise Error(ErrorCode.InvalidParams, "expected params to be a dictionary")
+        if "method" not in params:
+            raise Error(ErrorCode.InvalidParams, 'expected "method" key')
+        method = params["method"]
+        if not isinstance(method, str):
+            raise Error(ErrorCode.InvalidParams, 'expected "method" key to be a string')
+        self._notify(method, params.get('params'))
```

Comment:
This does not actually send out the notification before returning from the request. It schedules it on the runloop so it will be sent out later. If you want to be sure of this, you could make the `_log`, `_notify`, `_reply` and `_error` methods `async` and they can `await` on `_send_payload`. And then do `await self._notify(...)`. That would guarantee that the notification is sent before returning here.

If you don't care about this, then I'm okay with how it looks :)

Please update the comments at the top: remove the TODO about sending notifications and explain how to create fake notifications.

Reply:
> This does not actually send out the notification before returning from the request. It schedules it on the runloop so it will be sent out later.

True, but the response to the request is also scheduled on the runloop. So given the order they are scheduled, they should arrive in that order (request response after notification response), I would think.

Reply:
> True, but the response to the request is also scheduled on the runloop.

You're right.